lib/conncache.c

```diff
@@ -100,25 +124,38 @@ static void free_bundle_hash_entry(void *freethis)
   bundle_destroy(b);
 }
 
-int Curl_conncache_init(struct conncache *connc, size_t size)
+static curl_off_t next_cache_id;
```

Comment:
Without thinking through properly but is there not a risk that this misbehaves in multi-threaded contexts? Should it not be a field in the multi handle instead of being a global static?

Reply:
The problem are the shares with their conncache. Initially, they managed a shutdown list as well, but that got messy in the event based handling. So, we could remove the cache id again, I guess, and avoid this.

Reply:
cache id removed again.